spinalcordtoolbox/image.py

```diff
@@ -404,24 +404,22 @@ def set_qform_to_sform(self):
         self.hdr.set_qform(self.hdr.get_sform())
         self.hdr._structarr['qform_code'] = self.hdr._structarr['sform_code']
 
-    def fix_header_dtype(self, header='self', array='self'):
+    def fix_header_dtype(self):
         """
         Change the header dtype to the match the datatype of the array.
-
-        Note: This method defaults to operating on the image's `data` and `hdr` properties,
-              but the fix can also be applied to supplied arrays and headers, too."""
-        hdr = self.hdr if header == 'self' else header
-        data = self.data if array == 'self' else array
-        dtype_header = hdr.get_data_dtype()
-        dtype_data = data.dtype
-        if dtype_data != dtype_header:
-            if dtype_data == 'bool':
-                # Sometimes we create boolean arrays, apparently! But, using 'bool' for nibabel headers triggers:
-                # `nibabel.spatialimages.HeaderDataError: data dtype "bool" not supported`, so use `uint8` instead
-                dtype_data = np.uint8
+        """
+        # Using bool for nibabel headers is unsupported, so use uint8 instead:
+        # `nibabel.spatialimages.HeaderDataError: data dtype "bool" not supported`
```

Comment:
Thank you for updating this comment! My first draft of this was a little unprofessional, so I appreciate cleaning up the language here. :)

Reply:
I think your version was fine really, I guess I was just in a rewriting mood, sorry.

Reply:
No no no, oh gosh. I was actually feeling self-conscious about what I had written, even before your rewrite? 

I wanted to fix it up before you did, so I'm glad you did. :smile: